rest/route/admin_clear.go

```diff
@@ -0,0 +1,45 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/model/distro"
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/gimlet"
+)
+
+type clearTaskQueueHandler struct {
+	distro string
+	sc     data.Connector
+}
+
+func makeClearTaskQueueHandler(sc data.Connector) gimlet.RouteHandler {
+	return &clearTaskQueueHandler{sc: sc}
+}
+
+func (h *clearTaskQueueHandler) Factory() gimlet.RouteHandler { return &clearTaskQueueHandler{sc: h.sc} }
+
+func (h *clearTaskQueueHandler) Parse(ctx context.Context, r *http.Request) error {
+	vars := gimlet.GetVars(r)
+	h.distro = vars["distro"]
```

Comment:
Can you fix EVG-3628 since you're refactoring this? "distro" is supposed to be a query param instead